File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1

        while start <= end:
            curr = start + (end - start) // 2
            val = nums[curr]

            if val < target:
                start = curr + 1
            elif val > target:
                end = curr - 1
            else:
                return curr

        return -1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
   public:
    int longestConsecutive(vector<int>& nums) {
        int count = 0;
        int maxCount = 0;
        unordered_set<int> s(nums.begin(), nums.end());

        int i = 0;
        for (const auto& curr : s) {
            if (s.count(curr) && !s.count(curr-1)) {
                count = 1;
                int nextNum = curr + 1;
                while (s.count(nextNum)) {
                    nextNum++;
                    count++;
                }
                if (count > maxCount) {
                    maxCount = count;
                }
            }
        }

        return maxCount;
    }
};